docs/navigating.md

```diff
@@ -127,6 +126,17 @@ class DetailsScreen extends React.Component {
 
 Another common requirement is to be able to go back _multiple_ screens -- for example, if you are several screens deep in a stack and want to dismiss all of them to go back to the first screen. In this case, we know that we want to go back to `Home` so we can use `navigate('Home')` (not `push`! try that out and see the difference). Another alternative would be `navigation.popToTop()`, which goes back to the first screen in the stack.
 
+Navigation actions always behave idempotently. If you are not using `^2.0.0`, navigation actions behave idempotently only when called with a key. When pushing, the navigator will push a route with that key. If the route already exists, it will navigate to that route and set the params.
+
+```js
+<Button
```

Comment:
the text here is a little confusing:

> Go to Details... but don't push it to the stack if it already exists, just navigate

this is actually the behavior if you don't pass a key in, see: https://github.com/react-navigation/rfcs/blob/master/text/0004-less-pushy-navigate.md